code/script.js

```diff
@@ -1,19 +1,178 @@
+    pizzaType();
+  }
+  else if (food == 2) {
+    food = 'salad';
+    saladType();
+  }
+  else if (food == 3) {
+    food = "pasta";
+    pastaType();
+  }
+  //subType(food);
+}
+
+
+foodChoice();
 // Step 3 - Subtype choice
-// Your code goes here
+
+function pizzaType() {
+  order = prompt('Press 1 for Margarita,\n press 2 for Pineapple and ham \n press 3 for Diavola');
+
+  while ((order != 1) && (order != 2) && (order != 3)) {
+    order = prompt('Press 1 for Margarita,\n press 2 for Pineapple and ham \n press 3 for Diavola');
+  }
+  if (order == 1) {
+    order = 'Margarita ' + food;
+    food = order;
+    foodCost = 100;
+  }
+  else if (order == 2) {
+    order = 'Ham & Pineapple ' + food;
+    food = order;
+    foodCost = 150;
+  }
+  else if (order == 3) {
+    order = 'Diavola ' + food;
+    food = order;
+    foodCost = 180;
+  }
+  //}
+  finalOrder();
+}
+
+
+function saladType() {
+  order = prompt('Press 1 for Greek salad\npress 2 for Caesar salad \npress 3 for Watermelon salad');
+
+  while ((order != 1) && (order != 2) && (order != 3)) {
+    order = prompt('Press 1 for Greek salad\npress 2 for Caesar salad \npress 3 for Watermelon salad');
+  }
+
+  if (order == 1) {
+    order = 'Greek ' + food;
+    food = order;
+    foodCost = 75;
+  }
+  else if (order == 2) {
+    order = 'Caesar ' + food;
+    food = order;
+    foodCost = 90;
+  }
+  else if (order == 3) {
+    order = 'Watermelon ' + food;
+    food = order;
+    foodCost = 110;
+  }
+  finalOrder();
+}
+
+
+function pastaType() {
+  order = prompt('Press 1 for Penne pasta\n press 2 for Spaghetti \n press 3 for Gnocchi');
+
+  while ((order != 1) && (order != 2) && (order != 3)) {
+    order = prompt('Press 1 for Penne pasta\n press 2 for Spaghetti \n press 3 for Gnocchi');
+  }
+
+  if (order == 1) {
+    order = 'Penne ' + food;
+    food = order;
+    foodCost = 120;
+  }
+  else if (order == 2) {
+    order = 'Spaghetti';
+    food = order;
+    foodCost = 210;
+  }
+  else if (order == 3) {
+    order = 'Gnocchi';
+    food = order;
+    foodCost = 350;
+  }
+  finalOrder();
+}
+
 
 // Step 4 - Age
-// Your code goes here
+
+
+
+function finalOrder() {
+
+  age = prompt('Is the ' + food + ' for a child or an adult? \nPress 1 if its for a child\n Press 2 if its for an adult');
+
+  while ((age != 1) && (age != 2)) {
+    age = prompt('Is the ' + food + ' for a child or an adult? \nPress 1 if its for a child\nPress 2 if its for an adult');
+  }
+
+  if (age == 1) {
+    foodCostChild = foodCost * 0.80;
+    age = 'childrens';
+    foodCost = foodCostChild;
+    confirmOrder();
+  }
+  else if (age == 2) {
+    foodCost = foodCost;
+    age = 'regular';
+    confirmOrder();
+  }
+
+
+}
+
+
 
 // Step 5 - Order confirmation
 // Your code goes here
+
+function confirmOrder() {
+  let confirm = parseInt(prompt("You have ordered a " + age + " size " + food + ". \ncost: " + foodCost + " Bytecoin.\nPress 1 if this is correct. \nPress 2 if it's wrong."));
```

Comment:
this is correct, but have you thought about using template literals here to improve readability? :)

Reply:
good suggestion